packages/stark-ui/src/modules/date-picker/components/date-picker.component.spec.ts

```diff
@@ -1,171 +1,335 @@
-import { EventEmitter } from "@angular/core";
+import { Component, ViewChild } from "@angular/core";
 import { async, ComponentFixture, TestBed } from "@angular/core/testing";
 import { DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE } from "@angular/material/core";
 import { MatDatepickerModule } from "@angular/material/datepicker";
 import { MatInputModule } from "@angular/material/input";
 import { MAT_MOMENT_DATE_FORMATS, MatMomentDateModule, MomentDateAdapter } from "@angular/material-moment-adapter";
-import { TranslateModule, TranslateService } from "@ngx-translate/core";
+import { TranslateModule } from "@ngx-translate/core";
 import { STARK_LOGGING_SERVICE, STARK_ROUTING_SERVICE } from "@nationalbankbelgium/stark-core";
 import { MockStarkLoggingService, MockStarkRoutingService } from "@nationalbankbelgium/stark-core/testing";
-import { StarkDatePickerComponent } from "./date-picker.component";
+import { FormControl, FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { StarkDatePickerComponent, StarkDatePickerMaskConfig } from "./date-picker.component";
+import { StarkInputMaskDirectivesModule } from "../../input-mask-directives";
+import moment from "moment";
+
+/**
+ * To be able to test changes to the input fields, the Date Picker component is hosted inside the TestHostComponent class.
+ * This one uses the Output "dateChange"
+ */
+@Component({
+	selector: `host-component`,
+	template: `
+		<mat-form-field>
+			<stark-date-picker
+				[required]="required"
+				[disabled]="isDisabled"
+				[placeholder]="placeholder"
+				[value]="value"
+				(dateChange)="onValueChange($event)"
+			></stark-date-picker>
+		</mat-form-field>
+	`
+})
+class TestHostComponent {
+	@ViewChild(StarkDatePickerComponent)
+	public datePickerComponent: StarkDatePickerComponent;
+
+	public placeholder: string;
+	public value: Date;
+	public isDisabled: boolean;
+	public required: boolean;
+
+	/**
+	 * Simulates the OnValueChanges event of the date-picker component
+	 * To be able to test the 'Changes' output
+	 * @param value: Date.
+	 */
+	public onValueChange(value: Date): void {
+		this.value = value;
+	}
+}
+
+/**
+ * To be able to test changes to the input fields, the Date Picker component is hosted inside the TestHostFormControlComponent class.
+ * This one does not use the Output "dateChange"
+ */
+@Component({
+	selector: `host-form-control-component`,
+	template: `
+		<mat-form-field>
+			<stark-date-picker [required]="required" [dateMask]="dateMask" [formControl]="formControl"></stark-date-picker>
+		</mat-form-field>
+	`
+})
+class TestHostFormControlComponent {
+	@ViewChild(StarkDatePickerComponent)
+	public datePickerComponent: StarkDatePickerComponent;
+
+	public dateMask: StarkDatePickerMaskConfig;
+	public formControl: FormControl = new FormControl();
+	public required: boolean;
+}
 
 describe("DatePickerComponent", () => {
```

Comment:
Can you also add some unit tests for the dateMask? Just testing that the mask is enabled/disabled when it should depending on what is passed in the `dateMask` input 😉